### PR TITLE
Update setup.py, add exclude for test package

### PR DESCRIPTION
**Pull request by onkelbeh**
...otherwise it would try to install a package called 'test' at top level:

```
>>> Jobs: 0 of 1 complete, 1 failed                 Load avg: 0.87, 0.78, 0.73
 * Package:    dev-python/xknx-0.16.0
 * Repository: HomeAssistantRepository
 * Maintainer: b@edevau.net
 * Upstream:   julius@mittenzwei.com
 * USE:        abi_x86_64 amd64 elibc_glibc kernel_linux python_targets_python3_8 test userland_GNU
 * FEATURES:   network-sandbox preserve-libs sandbox userpriv usersandbox
>>> Unpacking source...
>>> Unpacking xknx-0.16.0.tar.gz to /var/tmp/portage/dev-python/xknx-0.16.0/work
>>> Source unpacked in /var/tmp/portage/dev-python/xknx-0.16.0/work
>>> Preparing source in /var/tmp/portage/dev-python/xknx-0.16.0/work/xknx-0.16.0 ...
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-python/xknx-0.16.0/work/xknx-0.16.0 ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-python/xknx-0.16.0/work/xknx-0.16.0 ...
 * python3_8: running distutils-r1_run_phase distutils-r1_python_compile
python3.8 setup.py build -j 10
running build
running build_py
running install_egg_info
running egg_info
writing xknx.egg-info/PKG-INFO
writing dependency_links to xknx.egg-info/dependency_links.txt
writing requirements to xknx.egg-info/requires.txt
writing top-level names to xknx.egg-info/top_level.txt
reading manifest file 'xknx.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'xknx.egg-info/SOURCES.txt'
Copying xknx.egg-info to /var/tmp/portage/dev-python/xknx-0.16.0/image/_python3.8/usr/lib/python3.8/site-packages/xknx-0.16.0-py3.8.egg-info
running install_scripts
 * ERROR: dev-python/xknx-0.16.0::HomeAssistantRepository failed (install phase):
 *   Package installs 'test' package which is forbidden and likely a bug in the build system.
 *
 * Call stack:
 *     ebuild.sh, line  125:  Called src_install
 *   environment, line 2972:  Called distutils-r1_src_install
 *   environment, line 1196:  Called _distutils-r1_run_foreach_impl 'distutils-r1_python_install'
```

## Description

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] The documentation has been adjusted accordingly
- [x] The changes generate no new warnings
- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] The changes are documented in the changelog
- [ ] The Homeassistant plugin has been adjusted in case of new config options

### CODE REVIEW

**Comment by farmio** [comment]
Why does this issue come up only now? This was not touched for ages...

Can we use `find_packages(where='xknx')` instead of excluding "test" to also exclude the HA custom component and any (future) scripts etc. in the base dir?

**Comment by onkelbeh** [comment]
I don't know. xknx-0.15.6 compiles fine.
Most others prefer an exclude, but I am no expert at this.
I have seen things like
```python
packages=find_packages(include=['xknx'])
```
somewhere.

Here I am using your module for [HomeAssistantRepository](https://github.com/onkelbeh/HomeAssistantRepository), hope you don't mind. It can be compiled with the patch at the bottom of https://github.com/onkelbeh/HomeAssistantRepository/blob/master/dev-python/xknx/xknx-0.16.0.ebuild, so I filed a PR the way I did it.

Thanks a lot.

File: setup.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
     ],
-    packages=find_packages(),
+    packages=find_packages(exclude=['test','test.*']),
```
Comment (farmio) [inline setup.py:40]:
```suggestion
    packages=find_packages(exclude=["test", "test.*"]),
```
black

**Review by farmio — CHANGES_REQUESTED** [review]

**Review by farmio — APPROVED** [review]

**Comment by farmio** [comment]
`packages=find_packages(include=["xknx", "xknx.*"]),` would indeed have the same result.

**Review by marvin-w — APPROVED** [review]
I'd prefer include over exclude as well. Looks fine though, thanks!

**Comment by farmio** [comment]
oh, it seems #557 would have fixed this too as it was a problem from setuptools

**Pull request merged**